src/sparsify/auto/tasks/object_detection/yolov5/runner.py

```diff
@@ -121,6 +123,105 @@ def config_to_args(
+        data files.
+
+        :returns: path to yaml to download or the newly built yaml. If the data string
+        arg is a yaml for a publicly available dataset, this function will return the
+        same string. Otherwise, the path to the newly generated yaml will be returned.
+        """
+        data_file_args = {}
+        IMAGE_DIR = "images"
+        CLASS_LIST_PATH = "classes.txt"
+        YAML_PATH = "data_local.yaml"
+
+        def _check_and_update_file(file_type: str, path: str):
+            if data_file_args.get(file_type, None):
+                data_file_args[file_type].append(path)
+            else:
+                data_file_args[file_type] = [path]
+
+        if os.path.isdir(dataset):
+            image_path = os.path.join(dataset, IMAGE_DIR)
+            class_list_path = os.path.join(dataset, CLASS_LIST_PATH)
+
+            if not os.path.exists(image_path):
+                raise ValueError(
+                    f"The the provided directory path {dataset} "
+                    "does not contain a folder called `images`. A subdirectory must "
+                    "exist which contains the data folders."
+                )
+
+            if not os.path.exists(class_list_path):
+                raise ValueError(
+                    f"The the provided directory path {dataset} "
+                    "does not contain a classes.txt file. A file must be "
+                    "present which includes a list of the classes for the dataset."
+                )
+
+            data_file_args["path"] = dataset
+
+            for d in os.listdir(image_path):
+                current_path = os.path.join(IMAGE_DIR, d)
+                if re.search(r"train", d):
+                    _check_and_update_file("train", current_path)
+                elif re.search(r"val", d):
+                    _check_and_update_file("val", current_path)
+                elif re.search(r"test", d):
+                    _check_and_update_file("test", current_path)
+
+            if not (
+                data_file_args.get("train", None) and data_file_args.get("val", None)
+            ):
+                raise Exception(
+                    "No training or validation folders found. Be sure the "
+                    "directory provided to the data arg contains folders "
+                    "with the train and val substrings in the filenames."
+                )
+
+        if data_file_args:
+            # Store the newly generated yaml in the same directory as the data
+            dataset = os.path.join(dataset, YAML_PATH)
+
+            with open(class_list_path, "r") as f:
+                class_list = f.readlines()
+
+            classes = {}
+            for class_id in range(len(class_list)):
+                classes[class_id] = class_list[class_id].strip()
+
+            with open(dataset, "w") as f:
+                yaml.safe_dump(
+                    {**data_file_args, **{"names": classes}}, f, sort_keys=False
```

Comment:
I don't think we need the inner dictionary here:

```python
>>> a = {"a": 1}
>>> b = {"b": 2, **a}
>>> b
{'b': 2, 'a': 1}
>>> c = {**a, "c": 3}
>>> c
{'a': 1, 'c': 3}
>>>
```

```suggestion
                    {**data_file_args, "names": classes}, f, sort_keys=False
```